pkg/provider/azure_utils_test.go

```diff
@@ -255,3 +258,59 @@ func TestGetServiceAdditionalPublicIPs(t *testing.T) {
 		})
 	}
 }
+
+func TestRemoveDuplicatedSecurityRules(t *testing.T) {
+	for _, testCase := range []struct {
+		description string
+		rules       []network.SecurityRule
+		expected    []network.SecurityRule
+	}{
+		{
+			description: "no duplicated rules",
+			rules: []network.SecurityRule{
+				{
+					Name: to.StringPtr("rule1"),
+				},
+				{
+					Name: to.StringPtr("rule2"),
+				},
+			},
+			expected: []network.SecurityRule{
+				{
+					Name: to.StringPtr("rule1"),
+				},
+				{
+					Name: to.StringPtr("rule2"),
+				},
+			},
+		},
+		{
+			description: "duplicated rules",
+			rules: []network.SecurityRule{
+				{
+					Name: to.StringPtr("rule1"),
+				},
+				{
+					Name: to.StringPtr("rule2"),
+				},
+				{
+					Name: to.StringPtr("rule1"),
+				},
+			},
+			expected: []network.SecurityRule{
+				{
+					Name: to.StringPtr("rule2"),
+				},
+				{
+					Name: to.StringPtr("rule1"),
+				},
+			},
+		},
+	} {
+		t.Run(testCase.description, func(t *testing.T) {
+			rules := testCase.rules
+			rules = removeDuplicatedSecurityRules(rules)
+			assert.Equal(t, testCase.expected, rules, testCase.description)
```

Comment:
I prefer no `testCase.description` here since L310 already includes it.

Reply:
done